src/qt/walletmodel.cpp

```diff
@@ -559,7 +559,7 @@ bool WalletModel::isWalletEnabled()
 
 bool WalletModel::privateKeysDisabled() const
 {
-    return m_wallet->IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS);
+    return m_wallet->IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS) && m_wallet->KeypoolCountExternalKeys() == 0;
```

Comment:
This function should be renamed or called into from a new one.

`WalletModel::KeypoolEmptyAndPrivkeysDisabled` or something obnoxiously explicit.

Reply:
Done